programs/fileio.c

```diff
@@ -1346,14 +1346,25 @@ FIO_compressZstdFrame(FIO_ctx_t* const fCtx,
 
     DISPLAYLEVEL(6, "compression using zstd format \n");
 
+    size_t pledgedSrcSize = ZSTD_CONTENTSIZE_UNKNOWN;
```

Comment:
This code base uses `C90` convention, which only allows declaration of variables at the beginning of blocks.

Depending on the lifetime of the variable, you may either add the variable declaration at the top of current block,
or create a new sub-block specifically for this variable.

The second method is actually pretty good whenever the variable has a short lifespan.
This is for example the case for `int windowLog` below.